peepingtom/__main__.py

```diff
@@ -1,14 +1,75 @@
-import click
-import napari
+import sys
+import argparse
+from IPython.terminal.embed import InteractiveShellEmbed
 
-import peepingtom
+import peepingtom as pt
 
 
-@click.command()
-@click.argument('paths', nargs=-1, type=click.Path(exists=True))
-@click.option('-f', '--force_mode', default=None)
-def cli(paths, force_mode=None, **kwargs):
-    """Command line entrypoint for the peep function
-    """
-    with napari.gui_qt():
-        return peepingtom.peep(paths, force_mode, **kwargs)
+def parse():
+    parser = argparse.ArgumentParser(formatter_class=argparse.RawDescriptionHelpFormatter,
+                                     description='''
+PeepingTom command line interface.
+
+Opens files in napari and lands in an interactive ipython shell
+with peepingtom imported as `pt` and the initialised Peeper available as `p`.
+''',
+                                     epilog='''
+mode choices:
+  - lone: each datablock in a separate volume
+  - zip_by_type: one of each datablock type per volume
+  - bunch: all datablocks in a single volume
+
+EXAMPLES
+
+Open a .star file as particles:
+  peep particles.star
+
+Open particles and images from a directory:
+  peep /dir/with/star_and_mrc_files/
+
+Match files such as MyProtein_10.star and MyProtein_001.mrc,
+and name the respective DataBlocks Protein_10 and Protein_001:
+  peep /path/to/dir/ -f Protein -n 'Protein_\d+'
+''')
+    parser.add_argument('paths', nargs='+',
+                        help='any number of file or directories')
+    parser.add_argument('-m', '--mode', choices=('lone', 'zip_by_type', 'bunch'),
+                        help='how to arrange DataBlock into volumes [default: guess from input]')
+    parser.add_argument('-f', '--filters',
+                        help='a regex used to select filenames [default: .*]')
+    parser.add_argument('-n', '--name-regex', metavar='regex',
+                        help='a regex used to infer DataBlock names from paths [fallback: \d+]')
+    parser.add_argument('-r', '--recursive', action='store_true',
+                        help='read directories recursively')
+    parser.add_argument('-d', '--dry-run', action='store_true',
+                        help='only show the list of files that would be read')
+    parser.add_argument('--strict', action='store_true',
+                        help='immediately fail if a matched filename cannot be read')
+    parser.add_argument('--max', type=int,
+                        help='max number of files to read')
+    parser.add_argument('-V', '--version', action='version', version=f'%(prog)s - PeepingTom {pt.__version__}')
```

Comment:
I don't mind the switch to argparse but I'm not sure we actually gain anything from it - napari already indirectly depends on click, I don't see you do anything click can't do with a docstring and annotations (https://click.palletsprojects.com/en/7.x/documentation/?highlight=help) and IMO it's less readable?
Functionality and help text are great though!

Reply:
Ok, dependency problem is not a problem then, good to know! Other stuff I was thingking:
- click doesn't let you set inline help text to non-option arguments, whick kinda bothers me
- ~click forces me to repeat the function signature 3 times instead of 3~ actually, turns out you can use `**kwargs` in the function definition and just have to declare the arguments 2 times just like argparse
- I do like having parsing and function call separated, if anything for testing purposes

So yeah, now that you mention it it's not that big of a difference, so we could switch back. I don't think readability changes much, either.

Reply:
I think I'll leave as is for now, we can go back to click in a future PR!